OPAL/br/src/main/scala/org/opalj/br/analyses/FieldAccessInformationKey.scala

```diff
@@ -26,6 +26,10 @@ object FieldAccessInformationKey extends ProjectInformationKey[FieldAccessInform
                     "analysis configuration",
                     s"no field access information analysis configured, using SimpleFieldAccessInformationAnalysis as a fallback"
                 )(project.logContext)
+                project.updateProjectInformationKeyInitializationData(this) {
+                    case None             => Seq(EagerSimpleFieldAccessInformationAnalysis)
+                    case Some(schedulers) => schedulers ++ Seq(EagerSimpleFieldAccessInformationAnalysis)
```

Comment:
This feels weird, since we'd still only return the EagerSimpleFieldAccessInformationAnalysis as the only requirement, so if there actually were other schedulers here, I suppose that could end up in a dead lock because of undeclared requirements.
Probably, this case should never happen anyway, because if initialization data was update while requirements are already computed, we could end up here with the compute method running different schedulers than the ones used here to compute the requirements.
Not sure what the best solution to this is. Keep yours? Replace this line with an exception? Replace the whole functionality here with getOrCreateProjectInformationKeyInitializationData?

Reply:
I would either just use `getOrCreateProjectInformationKeyInitializationData` or throw an exception when a `Some(...)` is found, as the result handling of `getProjectInformationKeyInitializationData` and the update should happen in sync.

I will try to prevent this entirely by using `getOrCreateProjectInformationKeyInitializationData`.

Reply:
It works, but we cannot give a compute function to this `getOrCreateProjectInformationKeyInitializationData` which means we lose the info log. Is that acceptable?

Reply:
Why? getOrCreateProjectInformationKeyInitializationData takes a by-name parameter, that should allow adding the logging?

Reply:
I think I just had a brainfart, forgot that you can pass block expressions to everything in Scala 🤦